### PR TITLE
Checkbox minor fixes docs

### PR DESCRIPTION
**Pull request by akomiqaia**
## Checklist before requesting a review

- [ ] I have applied `percy` label for o-[COMPONENT] or `chromatic` label for o3-[COMPONENT] on my PR before merging and after review. Find more details in [CONTRIBUTING.md](https://github.com/Financial-Times/origami/blob/main/CONTRIBUTING.md#pull-requests-and-visual-regression-tests)
- [ ] If it is a new feature, I have added thorough tests.
- [ ] I have updated relevant docs.
- [ ] I have updated relevant env variables in Doppler.

### CODE REVIEW

File: components/o3-form/README.md [hunk 1]
```diff
@@ -2,22 +2,175 @@
...
 
 o3-form provides UI form elements with consistent labelling, descriptions, and error styles and interfaces.
 
 ## Components
 
+### Form
+
+Form is the top-level component that wraps all form elements. It can be used as plain HTML:
+
+```html
```
Comment (frshwtr) [inline components/o3-form/README.md:26]: The order of example code changes between each component here. In form , formfileds, and checkbox  it starts with HTML code followed by TSX, in text-input it changes to start with TSX and then HTML.

Can these be aligned so they have the same order of code examples?

File: components/o3-form/README.md [hunk 2]
```diff
@@ -2,22 +2,175 @@
...
+
+| Prop          | Description                                                                                        | required | Type   | Default |
+| ------------- | -------------------------------------------------------------------------------------------------- | -------- | ------ | ------- |
+| `label`       | The label for the form fieldset                                                                    | true     | string | -       |
+| `inputId`     | The id of the form input element                                                                   | false    | string | -       |
+| `description` | A description of the form fieldset                                                                 | false    | string | -       |
+| `children`    | The form elements to be rendered                                                                   | true     | any    | -       |
+| `feedback`    | The feedback object for the form fieldset that contains `message` and `type` of message properties | false    | string | -       |
+
+Form Fieldset is used to group related form elements together. They can be used as plain HTML:
```
Comment (frshwtr) [inline components/o3-form/README.md:126]: Add some examples of form elements that are normally grouped together.
```suggestion
Form Fieldset is used to group related form elements together such as checkboxes. They can be used as plain HTML:
```

**Comment by frshwtr** [comment]
This looks good with changes suggested. 

One question, is there a demo for intermediate check boxes?

File: components/o3-form/README.md (hunk 1 above)
Comment (akomiqaia) [reply components/o3-form/README.md:26]: [e0a5a9a](https://github.com/Financial-Times/origami/pull/1800/commits/e0a5a9a2846657e6b0bf95e46142ecc3daccd153)

**Comment by akomiqaia** [comment]
> This looks good with changes suggested.
> 
> One question, is there a demo for intermediate check boxes?

no, we don't. indeterminate can be set only with javascript and we will need to implement custom control on the storybook. do you think we should have it?

**Comment by frshwtr** [comment]
> no, we don't. indeterminate can be set only with javascript and we will need to implement custom control on the storybook. do you think we should have it?

If we can get some very small code to enable it, it will help to show users all the possible states. Due to the nature of the button, the readme would also need updating so they know they will need to implement their own JS to enable it.

**Comment by akomiqaia** [comment]
> If we can get some very small code to enable it, it will help to show users all the possible states. Due to the nature of the button, the readme would also need updating so they know they will need to implement their own JS to enable it.

I am happy to add this but maybe this goes against our progressive enhancement idea? I started [a thread](https://financialtimes.slack.com/archives/CSW6B2VAN/p1724770613266409) on origami chat channel for more details

**Comment by akomiqaia** [comment]
@frshwtr [6ea77e8](https://github.com/Financial-Times/origami/pull/1800/commits/6ea77e8d01bead4bbc1347834d2a3c6b9ecb8c9f) adds demo and improves readme

**Review by frshwtr — APPROVED** [review]

**Pull request merged**